### ISSUE TITLE
unable to start rclone: timed out waiting for rclone to start

### ISSUE THREAD

**Issue by segator** [kopia/kopia#2573]
1/3 `snapshot create` execution I got `unable to start rclone: timed out waiting for rclone to start`
**Version**: kopia:20221110.0.161523
**Backend**:  rclone mouting gdrive.

command executed
```bash
docker run --rm --hostname $(hostname) \
-e KOPIA_PASSWORD=xxxxxxxxxxx \
    -v /mnt/user/appdata/kopia/config:/app/config \
    -v /mnt/user/appdata/kopia/cache:/app/cache \
    -v /mnt/user/appdata/kopia/logs:/app/logs \
    -v /mnt:/mnt \
    -v /boot/config/plugins/rclone/.rclone.conf:/app/rclone/rclone.conf \
kopia/kopia:20221110.0.161523 snapshot create \
/mnt/user/appdata
```

**Comment by segator** [comment]
btw I notice this docker run return status code 0, it should return non 0, as it crashes when snapshoting

**Comment by marcosvrs** [comment]
Same here.
Works fine with version 0.11.3.

**Comment by cmerkle** [comment]
I have this same issue and it's very frustrating. Kopia will not automatically start when I reboot my server because it times out waiting for rclone to start, so after each reboot I have to manually run sudo systemctl start kopia. I did not manually set rclone to log to a file as mentioned above, I am not even sure how to do so. Can someone please help solve this issue?

**Comment by J-PC** [comment]
> Tried a lot and figured out something:
> 
> 1. If you set rclone log to file, either by `--rclone-env=` or `--rclone-args=` or any other method, then kopia will NOT able to detect rclone's state, it will always said time out. This is because kopia currently detect rclone by specific string in rclone's stderr stream, but rclone will write that to log file instead of stderr stream.
> 2. Depending on network conditions, sometimes rclone will need SERVERAL MINUTES to get to the stage 'started', and sometimes rclone seems need to finish unfinished upload jobs before 'started'(not confirmd). so I think kopia should use a much longer default timeout, or or give user an timeout option.
> 3. if rclone starts too fast, then kopia cant detect it too, because rclone get to the stage 'started' before kopia attach stderr. This should be rare.

Obviously, the first point about kopia using stderr to detect rclone is what prevents using systemd to launch kopia with an rclone backend. I get a time out error even though it works directly from the cli. I'm using a variant of the systemd service file with a timer from the following issue https://github.com/kopia/kopia/issues/2685#issuecomment-1384524828
It works with systemd user mode though.

The service file looks like :

```
[Unit]
Description=Backup
Wants=network-online.target
After=network-online.target

[Service]
Type=oneshot
ExecStart=/usr/bin/kopia snapshot create --all
PrivateTmp=true
StandardOutput=journal
StandardError=journal
```

I tried tinkering with the StandardError and StandardOutput switching to null but there is no effect.

**Comment by rusty-art** [comment]
Hey, just ran into this on Windows with kopia v0.22.3 and rclone v1.72.0 (OneDrive backend) and wanted to share what worked for me in case it helps others. 

The issue: kopia repository sync-to rclone was timing out after 15 seconds, even on a fast machine with no network issues. rclone worked perfectly when run standalone. 

## What fixed it: 
Adding --rclone-debug to the sync command:

kopia repository sync-to rclone --remote-path "remote:path" --rclone-debug

With the flag: instant detection, sync works perfectly. Without the flag: 100% timeout every time. 

Why it works: From what I can tell (at least in the older code - haven't checked if this changed recently), kopia detects rclone startup by parsing its stderr for specific strings. The --rclone-debug flag enables verbose rclone output which includes the patterns kopia looks for perhaps. Recent rclone versions may have changed their default output format, breaking detection. 

Bonus gotcha: If you've had multiple failed sync attempts, check for orphaned rclone processes - they can pile up and block new connections - that can be a related reason rclone is slow or timing out.

## Check for zombies
tasklist | findstr rclone

## Kill them (run as admin on Windows)
Get-Process rclone -ErrorAction SilentlyContinue | Stop-Process -Force

Hope this helps someone else who's experiencing this.
FWIW, I've got a set of Python wrapper scripts for automating kopia backups on Windows that now includes this work-around: github.com/rusty-art/kopia-helpers
